### PR TITLE
Added josephus problem recursive code in C++

### PR DESCRIPTION
**Pull request by abhishek7997**
Added recursive code for Josephus Problem in C++.
Issue #442

### CODE REVIEW

**Comment by abhishek7997** [comment]
@tarun26091999 Please review my PR.

**Review by tarun26091999 — CHANGES_REQUESTED** [review]
link it to the folder. also add the explanation at the top.

File: Code/C++/josephus_problem.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    josephus(lst, k, 0, ans);
+    cout << ans << endl;
+    return 0;
+}
+
+/*
+
+Test Case:
+
+Input: 3,2
```
Comment (yasharth291) [inline Code/C++/josephus_problem.cpp:47]: giving wrong output

**Review by yasharth291 — CHANGES_REQUESTED** [review]

File: Code/C++/josephus_problem.cpp (hunk 1 above)
Comment (abhishek7997) [reply Code/C++/josephus_problem.cpp:47]: I will check.

**Pull request closed without merging**